apps/files_sharing/lib/Controller/ShareAPIController.php

```
@@ -773,7 +776,7 @@ public function updateShare(
 			throw new OCSNotFoundException($this->l->t('Wrong share ID, share doesn\'t exist'));
 		}
 
-		if ($permissions === null && $password === null && $sendPasswordByTalk === null && $publicUpload === null && $expireDate === null && $note === null) {
+		if ($permissions === null && $password === null && $sendPasswordByTalk === null && $publicUpload === null && $expireDate === null && $note === null && $hideDownload === null) {
```

Any reason we force setting it and not just have a default of 'false' ?

---

We do not force setting it. This is just if nothing is set.

We can't default to false because we do not require to send all the parameters. If it would default to false it would always turn of hide download if for exmple you only change the xpiration date ;)

---

:see_no_evil: My mind somehow messed up the if clause, nevermind.